pandas/tests/reshape/test_pivot.py

```diff
@@ -2769,3 +2769,18 @@ def test_unstack_copy(self, m):
         result = df.unstack(sort=False)
         result.iloc[0, 0] = -1
         tm.assert_frame_equal(df, df_orig)
+
+    @pytest.mark.parametrize("tz", [None, "UTC"])
+    def test_pivot_empty_with_datetime(self, tz):
+        # GH#59126
+        df = DataFrame(
+            {
+                "timestamp": [pd.Timestamp("2024-01-01T00:00", tz=tz)],
```

Comment:
If your create the columns with `pd.Series(None, dtype=...)` you should get an empty `df` with the dtypes already (without having to do that filter in the next line)

Reply:
Yes, thanks!

Reply:
Actually I tried and keep getting the error before even testing:
https://github.com/pandas-dev/pandas/actions/runs/9713703751/job/26811065289

I guess we should also try to fix the construction of empty Series, but this is not the bug I'm after here.

Reply:
> "timestamp": pd.Series([], pd.DatetimeTZDtype(tz="UTC")),

The second argument positionally is `index`, not `dtype`

Reply:
Riiight, sorry I was sloppy.